lib/measured/measurable.rb

```diff
@@ -46,27 +43,14 @@ def inspect
   end
 
   def <=>(other)
-    if other.is_a?(self.class)
-      other_converted = other.convert_to(unit)
-      value <=> other_converted.value
-    elsif other == 0
-      other_converted = self.class.new(0, unit)
-      value <=> other_converted.value
+    case other
+    when self.class
+      value <=> other.convert_to(unit).value
```

Comment:
Yep. `convert_to` should be fast when it's already in the given unit, and now that things are immutable we don't have to worry about weird things when returning `self` from it :)